gene/pointMutation.go

```diff
@@ -0,0 +1,129 @@
+package gene
+
+import (
+	"github.com/vertgenlab/gonomics/dna"
+	"github.com/vertgenlab/gonomics/numbers"
+)
+
+// PointMutation changes a single nucleotide to the desired base, predicts the effect,
+// and updates the Gene struct to reflect the change.
+// The position of the mutation should be given in base-zero genomic coordinates.
+func PointMutation(g *Gene, genomePos int, alt dna.Base) (EffectPrediction, error) {
+	var answer EffectPrediction
+	answer.StopDist = -1
+	var err error
+
+	err = pointPreRunChecks(g, genomePos, &alt)
+	if err != nil {
+		return EffectPrediction{}, err
+	}
+
+	genomeIndexPos := numbers.AbsInt(genomePos - g.startPos) // abs needed to handle negative posStrand
```

Comment:
I am having trouble following this conversion.  I think we are changing an index where 0 is the first base of the chrom, to the corresponding index where 0 is the first base of the cDNA.  Now that I am typing this, I think what is throwing me off is that you are assuming that genomePos is within the gene, but I think I would feel more comfortable if we check that.  Correctly if they give a genomePos on the "wrong side" of the cDNA start, it will just fix it for them and mutate the position the same amount to the "right side" of the cDNA start.  While this works perfectly when the input is reasonable, it makes me nervous that unreasonable input is converted to something reasonable without an error.